lib/scripts/mtw.js

```diff
@@ -292,6 +301,36 @@ export class ContentScript {
   }
 
   /**
+   * Translate one word in each sentence for a  paragraph.
+   * @param {Object} paragraph - Paragraph nodeType
+   * @param {Object} filteredTMap - filtered translation map
+   * @param {Object} iMap - HTML element for each translated word
+   */
+  translateOneWord(paragraph, filteredTMap, iMap) {
+    for (let i in paragraph.childNodes) {
+      if (paragraph.childNodes[i].nodeType === 3) {
+        if(!/^\s*$/.test(paragraph.childNodes[i].textContent)){
+          let sentences = paragraph.childNodes[i].textContent.split('.');
+          for (let j in sentences) {
+            let words = sentences[j].split(' ');
+            words = _.shuffle(words);
+            for (let k in words) {
+              if (filteredTMap[words[k]]) {
+                let x = sentences[j].replace(words[k], iMap[filteredTMap[words[k]]]);
+                sentences[j] = x;
+                break;
+              }
+            }
+          }
+          var newNode = document.createElement('span');
+          newNode.innerHTML = sentences.join('.');
```

Comment:
@Ceilican we don't know which word will be translated. I am shuffling the words and and replacing the the first word whose translation is defined. This is equivalent (or rather almost equivalent) to randomly replacing one word in a sentence. 

Also, splitting words in a sentence may introduce errors due to rogue spaces. Hence I chose to sentence wise.